src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -0,0 +1,180 @@
+from sentry.seer.anomaly_detection.types import ADConfig, Alert, StoreDataRequest, TimeSeriesPoint
+from sentry.seer.signed_seer_api import make_signed_seer_api_request
+from sentry.snuba.models import SnubaQuery
+from sentry.snuba.referrer import Referrer
+from sentry.snuba.utils import get_dataset
+from sentry.utils import json
+from sentry.utils.snuba import SnubaTSResult
+
+logger = logging.getLogger(__name__)
+
+seer_anomaly_detection_connection_pool = connection_from_url(
+    settings.SEER_ANOMALY_DETECTION_URL,
+    timeout=settings.SEER_ANOMALY_DETECTION_TIMEOUT,
+)
+
+
+def format_historical_data(data: SnubaTSResult) -> list[TimeSeriesPoint]:
+    """
+    Format Snuba data into the format the Seer API expects.
+    If there are no results, it's just the timestamp
+    {'time': 1719012000}, {'time': 1719018000}, {'time': 1719024000}
+
+    If there are results, the count is added
+    {'time': 1721300400, 'count': 2}
+    """
+    formatted_data = []
+    for datum in data.data.get("data", []):
+        ts_point = TimeSeriesPoint(timestamp=datum.get("time"))
+        if datum.get("count"):
+            ts_point["value"] = datum.get("count")
+        else:
+            ts_point["value"] = 0
+
+        formatted_data.append(ts_point)
+    return formatted_data
+
+
+def is_data_empty(data: list[TimeSeriesPoint]) -> bool:
+    for datum in data:
+        if datum.get("value") != 0:
+            return False
+    return True
+
+
+def translate_direction(direction: int) -> str:
+    """
+    Temporary translation map to Seer's expected values
+    """
+    direction_map = {
+        AlertRuleThresholdType.ABOVE: "up",
+        AlertRuleThresholdType.BELOW: "down",
+        AlertRuleThresholdType.ABOVE_AND_BELOW: "both",
+    }
+    return direction_map[AlertRuleThresholdType(direction)]
+
+
+def get_project_id_from_rule(rule: AlertRule) -> int | None:
+    project = None
+    subscriptions = rule.snuba_query.subscriptions.all()
+    if subscriptions:
+        project = subscriptions[0].project
+
+        if not project:
+            return None
+    return project.id
+
+
+def send_historical_data_to_seer(rule: AlertRule, user: User) -> None:
+    if not features.has("organizations:anomaly-detection-alerts", rule.organization, actor=user):
+        return None
+
+    project_id = get_project_id_from_rule(rule)
+    if not project_id:
+        logger.error(
+            "No project associated with rule. Skipping sending historical data to Seer",
+            extra={
+                "rule_id": rule.id,
+                "project_id": project_id,
+            },
+        )
+        return None
+    snuba_query = SnubaQuery.objects.get(id=rule.snuba_query_id)
+    time_period = int(snuba_query.time_window / 60)
+    historical_data = fetch_historical_data(rule, snuba_query)
+    formatted_data = format_historical_data(historical_data)
+
+    if is_data_empty(formatted_data):
+        return BaseHTTPResponse(
```

Comment:
I'm using this here (docs: https://urllib3.readthedocs.io/en/latest/reference/urllib3.response.html) because it matches the return type used by `make_signed_seer_api_request` so it's consistent. It requires all these things are passed even though we only care about the `status` and `reason` here.